optimum/onnxruntime/optimization.py

```diff
@@ -54,22 +55,23 @@ def from_pretrained(
         Returns:
             An instance of `ORTOptimizer`.
         """
-        tokenizer = AutoTokenizer.from_pretrained(model_name_or_path)
+        preprocessor = get_preprocessor(model_name_or_path)
```

Comment:
Actually it is from what I see [here](https://github.com/huggingface/transformers/blob/b6a65ae52a6df67180be8f1e162a3d1829a51bc0/src/transformers/onnx/utils.py#L71).

Regarding a unified type called `Preprocessor`, I'm in favour of it but on the other hand it is not clear in Transformers either and we also have a different kind of preprocessor for quantization. So I feel like it will always be quite ambiguous.

Reply:
I agree with you on the second point, maybe explicit is better for the end-user. For the first point, I think we need to figure it out!

Reply:
Actually `AutoFeatureExtractor` and `AutoProcessor` will both return a `FeatureExtractorMixin` object.